integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/partition/TestDDLForPartitionTable.scala

```diff
@@ -122,6 +122,254 @@ class TestDDLForPartitionTable  extends QueryTest with BeforeAndAfterAll {
     intercept[Exception] { sql("alter table test drop columns(c)") }
   }
 
+  test("test exception if hash number is invalid") {
+    sql("DROP TABLE IF EXISTS test_hash_1")
+    try {
+      sql(
+        """
+          | CREATE TABLE test_hash_1(col1 INT, col2 STRING)
+          | PARTITIONED BY (col3 INT) STORED BY 'carbondata'
+          | TBLPROPERTIES('PARTITION_TYPE'='HASH', 'NUM_PARTITIONS'='2.1')
+        """.stripMargin
+      )
+    } catch {
+      case e: Exception =>
+        assert(e.getMessage.contains("Invalid partition definition"))
+    }
+
+    sql("DROP TABLE IF EXISTS test_hash_2")
```

Comment:
One test for failure validation is sufficient. As all of them are covered in parse exception